File: 算法/14. 最长公共前缀/LongestCommonPrefix.java
public class LongestCommonPrefix {

    class Solution {
        public String longestCommonPrefix(String[] strs) {
            if (strs == null || strs.length == 0) {
                return "";
            }
            String prefix = strs[0];
            for (String str : strs) {
                while (!str.startsWith(prefix)) {
                    if (prefix.length() == 1) {
                        return "";
                    }
                    prefix = prefix.substring(0,prefix.length()-1);
                }
            }
            return prefix;
        }
    }

}

File: 算法/14. 最长公共前缀/LongestCommonPrefix.go
package _4__最长公共前缀

import "strings"

func longestCommonPrefix(strs []string) string {
	first := strs[0]
	for j := 1; j < len(strs); j++ {
		index := strs[j]
		for !strings.HasPrefix(index, first) {
			if len(first) == 1 {
				return ""
			}
			first = first[0 : len(first)-1]
		}
	}
	return first
}
